### PR TITLE
AlertingNG: Unified alerting API mock

### PR DESCRIPTION
**Pull request by papagian**
**What this PR does / why we need it**:
Provide mock implementations of the remaining routes of unified alerting API.
This API implementation is for assisting frontend development and it will be removed as soon as the actual API implementation will be completed.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
requirers grafana/alerting-api#29

### CODE REVIEW

File: pkg/services/ngalert/api/api_testing_mock.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	mock.log.Info("RouteTestRuleConfig: ", "body", body)
+	result := apimodels.TestRuleResponse{
+		GrafanaAlertInstances: apimodels.AlertInstancesResponse{
+			Instances: [][]byte{
+				[]byte("QVJST1cxAAD/////+AAAABAAAAAAAAoADgAMAAsABAAKAAAAFAAAAAAAAAEDAAoADAAAAAgABAAKAAAACAAAAFAAAAACAAAAKAAAAAQAAACE////CAAAAAwAAAAAAAAAAAAAAAUAAAByZWZJZAAAAKT///8IAAAADAAAAAAAAAAAAAAABAAAAG5hbWUAAAAAAQAAABgAAAAAABIAGAAUAAAAEwAMAAAACAAEABIAAAAUAAAAQAAAAEQAAAAAAAAGQAAAAAEAAAAMAAAACAAMAAgABAAIAAAACAAAAAwAAAAAAAAAAAAAAAQAAABuYW1lAAAAAAAAAAAEAAQABAAAAAAAAAAAAAAA/////4gAAAAUAAAAAAAAAAwAFgAUABMADAAEAAwAAAAIAAAAAAAAABQAAAAAAAADAwAKABgADAAIAAQACgAAABQAAAA4AAAAAQAAAAAAAAAAAAAAAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAgAAAAAAAAAAAAAAAEAAAABAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAQAAAADAAUABIADAAIAAQADAAAABAAAAAsAAAAPAAAAAAAAwABAAAACAEAAAAAAACQAAAAAAAAAAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAKAAwAAAAIAAQACgAAAAgAAABQAAAAAgAAACgAAAAEAAAAhP///wgAAAAMAAAAAAAAAAAAAAAFAAAAcmVmSWQAAACk////CAAAAAwAAAAAAAAAAAAAAAQAAABuYW1lAAAAAAEAAAAYAAAAAAASABgAFAAAABMADAAAAAgABAASAAAAFAAAAEAAAABEAAAAAAAABkAAAAABAAAADAAAAAgADAAIAAQACAAAAAgAAAAMAAAAAAAAAAAAAAAEAAAAbmFtZQAAAAAAAAAABAAEAAQAAAAAAAAAAAAAACgBAABBUlJPVzE="),
```
Comment (owen-d) [inline pkg/services/ngalert/api/api_testing_mock.go:27]: What's going on here?

File: pkg/services/ngalert/api/util.go [hunk 2]
```diff
@@ -13,3 +15,59 @@ func toMacaronPath(path string) string {
...
+
+func stringPtr(s string) *string {
+	return &s
+}
+
+func boolPtr(b bool) *bool {
+	return &b
+}
+
+func anyPtr(v interface{}) *interface{} {
```
Comment (owen-d) [inline pkg/services/ngalert/api/util.go:71]: This seems unused and what's the need for `*interface{}`?

**Review by owen-d — APPROVED** [review]
A few comments, but it looks great. Thanks for taking care of this.

File: pkg/services/ngalert/api/api_testing_mock.go (hunk 1 above)
Comment (papagian) [reply pkg/services/ngalert/api/api_testing_mock.go:27]: Instances is an array of arrow encoded [dataframes](https://github.com/grafana/grafana/blob/0c2045109eabbce1abf17d679014a1b32879723c/pkg/plugins/dataframes.go#L15)

File: pkg/services/ngalert/api/util.go (hunk 2 above)
Comment (papagian) [reply pkg/services/ngalert/api/util.go]: None, I just copied and pasted it the others 😳

**Pull request merged**